terraform/node_provider.go

```diff
@@ -48,6 +48,13 @@ func (n *NodeApplyableProvider) ValidateProvider(ctx EvalContext, provider provi
 
 	configBody := buildProviderConfig(ctx, n.Addr, n.ProviderConfig())
 
+	// if provider config is empty, return early
+	emptySchema := &configschema.Block{}
+	_, _, evalDiags := ctx.EvaluateBlock(configBody, emptySchema, nil, EvalDataForNoInstanceKey)
```

Comment:
Silly Terraform tricks: another way to do this (the approach here is totally valid!)

```
_, noConfigDiags := configBody.Content(&hcl.BodySchema{})
if !noConfigDiags.HasErrors() {
	return nil
}
```

This is more to share, and offer if you like it, but your way works absolutely fine.